dev-libs/smooth/smooth-0.8.74.0~pre4.recipe

```diff
@@ -0,0 +1,95 @@
+SUMMARY="A cross-platform C++ class library"
+DESCRIPTION="smooth is an object oriented C++ class library for Windows, \
+macOS, Linux and most Unix-like operating systems. It provides basic \
+functionality and platform support for applications and libraries.
+
+Features provided by smooth include:
+
+- user interface API with various widgets
+- simple to use multithreading API
+- file and network IO interface
+- completely transparent Unicode and software internationalization support
+- a libxml2 based XML parser"
+HOMEPAGE="http://www.smooth-project.org/"
+COPYRIGHT="1998-2018 Robert Kausch"
+LICENSE="Artistic v2.0"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/smooth/smooth-${portVersion/\~/-}.tar.gz"
+CHECKSUM_SHA256="815f0329b5546c78feeb9ad1813855dfcb7bc062e28de44d0b4aadd7c6625d80"
+SOURCE_DIR="smooth-${portVersion/\~/-}"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+fi
+
+libVersion="${portVersion%~*}"
+libVersionCompat="$libVersion compat >= ${libVersion%.*}"
+
+PROVIDES="
+	smooth$secondaryArchSuffix = $portVersion
+	lib:libsmooth_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	"
+if [ -z "$commandSuffix" ]; then
+PROVIDES+="
+	cmd:smooth_translator = $portVersion
+	"
+fi
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	smooth${secondaryArchSuffix}_devel = $portVersion
+	devel:libsmooth_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	smooth$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libbz2$secondaryArchSuffix
+	devel:libcurl$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	cmd:g++$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make prefix=$prefix bindir=$binDir libdir=$libDir includedir=$includeDir install
```

Comment:
> > ```diff
> > -	make prefix=$prefix bindir=$binDir libdir=$libDir includedir=$includeDir install
> > +	make prefix=$prefix bindir=$commandBinDir libdir=$libDir includedir=$includeDir install
> > ```
>
> What's the point of defining a $commandBinDir over using $binDir directly?

Good catch. What we need here is:
```diff
-	make prefix=$prefix bindir=$binDir libdir=$libDir includedir=$includeDir install
+	make prefix=$prefix bindir=$prefix/bin libdir=$libDir includedir=$includeDir install
```
We used to define `commandBinDir` as `$binDir` on all arches, except on x86 secondary arch, where we define it as `$prefix/bin`.
(On x86 secondary arch `$binDir` is `$prefix/bin/x86` but we don't want to use it because we want the runtime to be in `$prefix/bin` in all cases, that is, even if we're build for x86 secondary arch.)

Now, given that:
1) smooth is not available on x86_gcc2 primary arch
2) smooth_x86 has its `$prefix/bin` removed on all arches except when `$commandSuffix` empty,

we don't need to define `commandBinDir` anymore. We can use `$prefix/bin` in all cases.

Reply:
@fbrosson would it be prefered to use the qoutes here too? eg "$prefix" etc

Reply:
I know some contributors argue that putting quotes around these variables is not needed, but I personally prefer to use them because they **_could_** be useful in some non common configurations: if a user decides to insert a space in the path that is used as the `TREE_PATH` in `haikuports.conf` and then runs `haikuporter --nochroot --test` for a recipe which uses e.g. `$sourceDir` in `TEST`, or chooses to cross-build a `"any"`-arch package which uses e.g. `$prefix` or `$dataDir`, then the build will fail if we don't have the quotes.

So, instead of having the quotes only on those recipes which could require them, I'd rather use them in all new recipes and in all recipes we update for some other reason.

That's why I recommend using them in e.g. `"$prefix"`, `"$dataDir"`, `"$binDir"`,`"$commandBinDir"`,`"$libDir"`, `"$includeDir"`, etc...

Moreover, even if `--nochroot` is currently only usable with `--test`, nobody can claim it won't ever be usable for building recipes.

N.B. I don't use quotes for `relativeBinDir`and other similar `relative*Dir`variables because these are relative to `$prefix` and will never contain any spaces.
That said, I don't object using quotes for them too. It's just that I think its easier to have the quotes adopted for the non-relative variables rather than for the non-relative variables and the relative variables.